sql/hive/src/main/java/org/apache/hadoop/hive/ql/io/orc/VectorizedSparkOrcNewRecordReader.java

```diff
@@ -0,0 +1,318 @@
+      } else {
+        return (byte)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public short getShort(int ordinal) {
+      LongColumnVector col = (LongColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return (short)col.vector[0];
+      } else {
+        return (short)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public int getInt(int ordinal) {
+      LongColumnVector col = (LongColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return (int)col.vector[0];
+      } else {
+        return (int)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public long getLong(int ordinal) {
+      LongColumnVector col = (LongColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return (long)col.vector[0];
+      } else {
+        return (long)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public float getFloat(int ordinal) {
+      DoubleColumnVector col = (DoubleColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return (float)col.vector[0];
+      } else {
+        return (float)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public double getDouble(int ordinal) {
+      DoubleColumnVector col = (DoubleColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return (double)col.vector[0];
+      } else {
+        return (double)col.vector[rowId];
+      }
+    }
+
+    @Override
+    public Decimal getDecimal(int ordinal, int precision, int scale) {
+      DecimalColumnVector col = (DecimalColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        return Decimal.apply(col.vector[0].getHiveDecimal().bigDecimalValue(), precision, scale);
+      } else {
+        return Decimal.apply(col.vector[rowId].getHiveDecimal().bigDecimalValue(),
+          precision, scale);
+      }
+    }
+
+    @Override
+    public UTF8String getUTF8String(int ordinal) {
+      BytesColumnVector bv = ((BytesColumnVector)columns[columnIDs.get(ordinal)]);
+      if (bv.isRepeating) {
+        return UTF8String.fromBytes(bv.vector[0], bv.start[0], bv.length[0]);
+      } else {
+        return UTF8String.fromBytes(bv.vector[rowId], bv.start[rowId], bv.length[rowId]);
+      }
+    }
+
+    @Override
+    public byte[] getBinary(int ordinal) {
+      BytesColumnVector col = (BytesColumnVector)columns[columnIDs.get(ordinal)];
+      if (col.isRepeating) {
+        byte[] binary = new byte[col.length[0]];
+        System.arraycopy(col.vector[0], col.start[0], binary, 0, binary.length);
+        return binary;
+      } else {
+        byte[] binary = new byte[col.length[rowId]];
+        System.arraycopy(col.vector[rowId], col.start[rowId], binary, 0, binary.length);
+        return binary;
+      }
+    }
+
+    @Override
+    public CalendarInterval getInterval(int ordinal) {
+      throw new NotImplementedException();
```

Comment:
I think the hive artifact used by Spark does not support vectorisation for all the datatypes and thats why this is not being implemented. In future, upgrading hive in Spark might make it possible to add it. Can you add a comment about this ?